webhooks/framework.go

```diff
@@ -0,0 +1,55 @@
+package webhooks
+
+import (
+	"fmt"
+	jwt "github.com/dgrijalva/jwt-go"
+	"github.com/rancher/webhook-service/drivers"
+)
+
+// ConstructPayload constructs jwt url
+func ConstructPayload(input map[string]interface{}) (map[string]interface{}, error) {
+	if _, ok := input["driver"].(string); !ok {
+		return nil, fmt.Errorf("Driver of type string not provided\n")
+	}
+	driverId := input["driver"].(string)
+	driver := drivers.GetDriver(driverId)
+	if driver == nil {
+		return nil, fmt.Errorf("Driver %s is not registered\n", driverId)
+	}
+	payload, err := driver.ConstructPayload(input)
+	if err != nil {
+		return nil, fmt.Errorf("Error %v in driver %s for ConstructPayload\n", err, driver)
+	}
+	return payload, nil
+}
+
+//Execute does a lookup for driver and calls it
+func Execute(payload map[string]interface{}) (int, map[string]interface{}, error) {
+	jwtSigned := payload["url"]
+	token, err := jwt.Parse(jwtSigned.(string), func(token *jwt.Token) (interface{}, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodRSA); !ok {
+			return nil, fmt.Errorf("Unexpected signing method: %v\n", token.Header["alg"])
+		}
+		return drivers.PublicKey, nil
+	})
+
+	if err != nil || !token.Valid {
+		return 0, nil, fmt.Errorf("Error %v in token Jwt\n", err)
+	}
+
+	//key is of invalid type on checking err
+	if claims, ok := token.Claims.(jwt.MapClaims); ok {
+		if _, ok := claims["driver"].(string); !ok {
+			return 0, nil, fmt.Errorf("Driver not found after decode\n")
+		}
+		driverId := claims["driver"].(string)
```

Comment:
@cjellick  I've used the similar check `if _, ok := claims["driver"].(string); !ok {}` everywhere because both the things, if the key is present and if it's of a certain type are getting checked.